File: auth_zip.js
const fs = require('fs');
const path = require('path');
const readline = require('readline');

try {
  require.resolve('archiver');
  require.resolve('archiver-zip-encrypted');
} catch (e) {
  console.error("❌ 错误: 缺少必要的压缩库。");
  console.error("请先在当前目录下运行以下命令安装:");
  console.error("npm install archiver archiver-zip-encrypted");
  process.exit(1);
}

const archiver = require('archiver');
archiver.registerFormat('zip-encrypted', require('archiver-zip-encrypted'));

// 使用简单的 readline 来获取密码（为保证兼容性，暂时不隐藏输入内容）
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

const askPassword = (query) => {
  return new Promise((resolve) => {
    rl.question(query, (answer) => resolve(answer));
  });
};

async function main() {
  const authDir = path.join(__dirname, 'auth');
  const outputFilename = 'auth.zip';
  const innerZipName = 'data.bundle';

  if (!fs.existsSync(authDir)) {
    console.error(`错误: 未找到 auth 目录 (${authDir})`);
    rl.close();
    return;
  }

  const files = fs.readdirSync(authDir).filter(f => f.endsWith('.json'));
  if (files.length === 0) {
    console.error("错误: auth 目录下没有 .json 文件。");
    rl.close();
    return;
  }

  console.log(`找到 ${files.length} 个认证文件。`);
  
  const password = await askPassword("请输入 ZIP 加密密码: ");
  const confirmPassword = await askPassword("请再次确认密码: ");

  if (password !== confirmPassword) {
    console.error("❌ 错误: 两次输入的密码不一致！");
    rl.close();
    return;
  }

  if (!password) {
    console.error("❌ 错误: 密码不能为空。");
    rl.close();
    return;
  }

  // 关闭 readline 以防止程序挂起
  rl.close();

  console.log("正在进行第一层打包（隐藏文件名）...");

  try {
    // 1. 生成内部无密码 ZIP
    await new Promise((resolve, reject) => {
      const output = fs.createWriteStream(innerZipName);
      const archive = archiver('zip', { zlib: { level: 9 } });
      
      output.on('close', resolve);
      archive.on('error', reject);
      archive.pipe(output);
      
      for (const file of files) {
        archive.file(path.join(authDir, file), { name: `auth/${file}` });
      }
      archive.finalize();
    });

    console.log("正在进行第二层加密压缩...");

    // 2. 生成外部加密 ZIP
    await new Promise((resolve, reject) => {
      const output = fs.createWriteStream(outputFilename);
      const archive = archiver('zip-encrypted', {
        zlib: { level: 9 },
        encryptionMethod: 'zip20', // 标准 zip 加密 (兼容 Linux 下老版 unzip)
        password: password
      });
      
      output.on('close', resolve);
      archive.on('error', reject);
      archive.pipe(output);
      
      archive.file(innerZipName, { name: 'bundle.zip' });
      archive.finalize();
    });

    // 3. 清理临时文件
    if (fs.existsSync(innerZipName)) {
      fs.unlinkSync(innerZipName);
    }

    console.log(`\n✅ 成功！双层加密后的文件已保存为: ${path.resolve(outputFilename)}`);
    console.log("💡 现在外部只能看到一个 'bundle.zip'，看不到具体的 json 文件名了。");

  } catch (error) {
    console.error(`💥 压缩失败: ${error.message}`);
    if (fs.existsSync(innerZipName)) fs.unlinkSync(innerZipName);
  }
}

main();

File: auth_zip.py
import os
import sys
import getpass

try:
    import pyzipper
except ImportError:
    print("错误: 未安装 pyzipper 库。")
    print("请先运行: pip install pyzipper")
    sys.exit(1)

def create_auth_zip():
    auth_dir = os.path.join(os.getcwd(), "auth")
    output_filename = "auth.zip"

    if not os.path.exists(auth_dir):
        print(f"错误: 未找到 auth 目录 ({auth_dir})")
        return

    # 获取所有 json 文件
    files = [f for f in os.listdir(auth_dir) if f.endswith(".json")]
    
    if not files:
        print("错误: auth 目录下没有 .json 文件。")
        return

    print(f"找到 {len(files)} 个认证文件。")
    
    # 交互式输入密码
    password = getpass.getpass("请输入 ZIP 加密密码: ")
    confirm_password = getpass.getpass("请再次确认密码: ")

    if password != confirm_password:
        print("错误: 两次输入的密码不一致！")
        return

    print(f"正在压缩并加密为 {output_filename}...")
    
    temp_inner_zip = "data.bundle" # 内部打包文件
    try:
        # 第一层：将所有文件打包进一个临时的未加密包中，这样外部就看不到文件名了
        print("正在进行第一层打包（隐藏文件名）...")
        with pyzipper.ZipFile(temp_inner_zip, 'w', compression=pyzipper.ZIP_DEFLATED) as zf:
            for f in files:
                file_path = os.path.join(auth_dir, f)
                # 统一放进 inner_auth 目录
                zf.write(file_path, arcname=os.path.join("auth", f))

        # 第二层：将内部包进行加密压缩
        print(f"正在进行第二层加密压缩...")
        # 注意：这里改用兼容性最强的传统加密 (ZipCrypto)，以便 Linux 的 unzip 命令直接解压
        with pyzipper.ZipFile(output_filename, 'w', compression=pyzipper.ZIP_DEFLATED) as zf:
            zf.setencryption(pyzipper.Encryption.TRADITIONAL)
            zf.setpassword(password.encode('utf-8'))
            zf.write(temp_inner_zip, arcname="bundle.zip")
                
        print(f"\n✅ 成功！双层加密后的文件已保存为: {os.path.abspath(output_filename)}")
        print("💡 现在外部只能看到一个 'bundle.zip'，看不到具体的 json 文件名了。")
    except Exception as e:
        print(f"压缩失败: {str(e)}")
    finally:
        # 清理临时文件
        if os.path.exists(temp_inner_zip):
            os.remove(temp_inner_zip)

if __name__ == "__main__":
    create_auth_zip()
